pkg/proxy/util/utils.go

```diff
@@ -56,3 +73,70 @@ func ShouldSkipService(svcName types.NamespacedName, service *api.Service) bool
 	}
 	return false
 }
+
+// GetNodeAddresses list all match node IP addresses based on given cidr list.  Inject NetworkInterface for test purpose.
+// We expect the cidr list passed in is already validated.
+// Given `default-route`, it will the IP of host interface having default route.
+// Given an empty input `[]`, it will return `0.0.0.0/0` and `::/0` directly.
```

Comment:
Need to document the semantics wrt "larger than" addresses.

If I say ["1.2.3.4/32", "0.0.0.0/0"] will it return both or just 0 ?  Simpler is just to return both, I guess, which maybe conflicts with earlier guidance, but GIGO ?

Reply:
Also please document the fact that some callers need concrete IPs, not ranges, which is why this exists.

Reply:
> If I say ["1.2.3.4/32", "0.0.0.0/0"] will it return both or just 0 ? Simpler is just to return both, I guess, which maybe conflicts with earlier guidance, but GIGO ?

It will return just `0.0.0.0/0`, which is smarter.

Reply:
New comment looks like,

```
// GetNodeAddresses return all matched node IP addresses based on given cidr slice.  Inject NetworkInterface for test purpose.
// We expect the cidrs passed in is already validated.
// Given an empty input `[]`, it will return `0.0.0.0/0` and `::/0` directly.
// If multiple cidrs is given, it will return the minimal IP sets, e.g. given input `[1.2.0.0/16, 0.0.0.0/0]`, it will only return `0.0.0.0/0`.
// If you want concrete IPs, e.g. 1.2.3.4, then the input should be 1.2.3.4/32.
```

Reply:
> Also please document the fact that some callers need concrete IPs, not ranges, which is why this exists.

Good point, it's fixed now.